data/transactions/logic/assembler.go

```diff
@@ -885,17 +888,23 @@ func init() {
 	// WARNING: special case op assembly by argOps functions must do their own type stack maintenance via ops.tpop() ops.tpush()/ops.tpusha()
 }
 
-type lineErrorWrapper struct {
+type lineError struct {
 	Line int
 	Err  error
 }
 
-func (lew *lineErrorWrapper) Error() string {
-	return fmt.Sprintf(":%d %s", lew.Line, lew.Err.Error())
+func addLine(line int, err error) error {
+	return &lineError{Line: line, Err: err}
+}
+func newLineError(line int, msg string) error {
+	return addLine(line, errors.New(msg))
+}
+func fmtLineError(line int, format string, args ...interface{}) error {
+	return addLine(line, fmt.Errorf(format, args...))
```

Comment:
two functions look exceeding, maybe leave `fmtLineError` only?

Reply:
I'm ok with that.  I was unsure if calling Errorf() on a static string without any args to substitute in was considered bad style, so I wanted to give the option to avoid it.